modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -437,8 +436,10 @@ bool  PngDecoder::readData( Mat& img )
                         memcpy(frameNext.getPixels(), frameCur.getPixels(), imagesize);
 
                     compose_frame(frameCur.getRows(), frameRaw.getRows(), bop, x0, y0, w0, h0, mat_cur);
-                    if (delay_den < 1000)
+                    if (delay_den < 100)
```

Comment:
Please cite your source, there are several implementations: https://github.com/davidmz/apng-js/blob/e5272dcd6900cbea2c341ce259c206a162cedec8/src/library/parser.js#L64 https://github.com/FFmpeg/FFmpeg/blob/19c95ecbff84eebca254d200c941ce07868ee707/libavformat/apngdec.c#L259


I believe the spec says differently though: https://wiki.mozilla.org/APNG_Specification#%60fcTL%60:_The_Frame_Control_Chunk

There is a difference between storing things in an animation and giving values to delays for display. I think you should simply do:

```cxx
if (!delay_den)
    delay_den = 100;
m_animation.durations.push_back(cvRound(1000.*delay_num/delay_den));

Reply:
for example delaynum 1 delayden 24 it means 1000 / 24 40 ms
for example delaynum 4 delayden 100 it means 1000 / 100 * 4  40 ms

Reply:
when saving for example 80 ms 
delay num 80 delayden 1000 it means 1000 / 1000 * 80 80 ms